demo/server/src/index.ts

```diff
@@ -0,0 +1,74 @@
+'use strict';
+/**
+ * Module dependencies.
+*/
+import app from './app';
+import * as http from 'http';
+
+/**
+ * Get port from environment and store in Express.
+ */
+const port = normalizePort(process.env.PORT || 3000);
+app.set('port', port);
+
+/**
+ * Create HTTP server.
+ */
+var server = http.createServer(app);
+
+/**
+ * Listen on provided port,on all network interfaces.
+ */
+server.listen(port, onListening);
+server.on('error', onError);
+
+/**
+ * Normalize a port into a number,string,or false.
+ */
+function normalizePort(val: any): number | string | boolean {
+  let port = parseInt(val, 10);
+
+  if (isNaN(port)) {
+    //name pipe
+    return val;
+  }
+
+  if (port >= 0) {
+    return port;
+  }
+
+  return false;
+}
+
+/**
+ * Event listener for HTTP server "error" event.
+ */
+function onError(error: any) {
```

Comment:
Sadly there is no standard in that name as node in different versions using integers etc as types and actual error comes from low level library.
Done refactor for this anyway. It's more clear now.